mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6650,10 +6653,105 @@ struct ThreadwiseCopyRewritePattern
         SmallVector<Value, 8> destLowerIndices =
             layeredDestIndices[layeredDestIndices.size() - 1];
 
+        // XXX: we do not  implement the logic of toEmitOOBStoreCheck yet
+        bool toEmitOOBStoreCheckLogic = false;
+        SmallVector<unsigned, 2> oobStoreCheckDims;
+        // if (destLowerIndices.size() == 5) {
+        // for (unsigned iter = 0; iter < destTransform.getNumResults();
+        //     ++iter) {
+        // this code is for testing ,check dim 2
+        // if (iter == 2) {
+        //  toEmitOOBStoreCheckLogic = true;
+        //  oobStoreCheckDims.push_back(iter);
+        // }
+        //}
+        //}
+
         // Store to dest.
         // Issue scalar store.
-        b.create<StoreOp>(loc, convertedScalarValue, op.dest(),
-                          destLowerIndices);
+        if (toEmitOOBStoreCheckLogic) {
+          SmallVector<Value, 8> destLowerStoreOP;
+          SmallVector<Value, 8> destLowerStoreOOBOp;
```

Comment:
nit: `destLowerStoreOOBOp` -> `destLowerStoreOOBIndices` to have the same naming convention with load logic above.